README.md

```diff
@@ -97,6 +101,101 @@ Example map response:
 [ { "text": "upper_25", "value": 1}, { "text": "upper_75", "value": 2} ]
 ```
 
+### /metrics
+
+`POST /metrics`
+
+In `Panel > Queries` page. When configuring a query request using `Builder` mode, it will send the request to obtain the available metrics. The request body will carry the current metric and payload. In the `Builder` mode, if the `reloadMetric` value in the load configuration is true, the api will also be triggered when the value is modified / switched.
+
+Example request:
+```json
+{}
+```
+Or. 
+```json
+{
+  "metric": "DescribeMetricList",
+  "payload":{
+    "cloud": "cf6591c5dad211eaa22100163e120f6e",
+    "namespace": "MySQL",
+  }
+}
+```
+Example response:
+```json
+[{
+  "label": "Describe metric list", // Optional. If the value is empty, use the value as the label
+  "value": "DescribeMetricList", // The value of the option.
+  "payloads": [{ // Configuration parameters of the payload.
+    "label": "Namespace", // The label of the payload. If the value is empty, use the value as the label.
+    "name": "namespace", // The name of the payload. If the value is empty, use the name as the label.
+    "type": "select", // If the value is select, the UI of the payload is a radio box. If the value is multi-select, the UI of the payload is a multi selection box; if the value is input, the UI of the payload is an input box.
+    "placeholder": "Please select namespace", // Input box / selection box prompt information.
+    "reloadMetric": true, // Whether to overload the metrics API after modifying the value of the payload.
+    "options": [{ // If the payload type is select / multi-select, the list is the configuration of the option list.
+      "label": "acs_mongodb", // The label of the payload select option.
+      "value": "acs_mongodb", // The label of the payload value.
+    },{
+      "label": "acs_rds",
+      "value": "acs_rds",
+    }]
+  },{
+    "name": "metric",
+    "type": "select"
+  },{
+    "name": "instanceId",
+    "type": "select"
+  }]
+},{
+  "value": "DescribeMetricLast",
+  "payloads": [{
+    "name": "namespace",
+    "type": "select"
+  },{
+    "name": "metric",
+    "type": "select"
+  },{
+    "name": "instanceId",
+    "type": "multi-select"
+  }]
+}]
+```
+The display is as follows:
+![Metrics in builder mode](./docs/images/builder-metrics.png)
+
+### /options
```

Comment:
Do I understand correctly that based on the API response, this will create custom UI? That said, any user can implement any selects with any options?

Reply:
> Do I understand correctly that based on the API response, this will create custom UI? That said, any user can implement any selects with any options?

I'm sorry I didn't understand what you meant.

When using the builder mode, the data source plug-in will obtain the metric configuration list through the "metrics" API. Each metric configuration will describe how to render the UI of the metric's payload (within the specification, of course).

For example, I have a metric named getrdsslowlogs, which is used to obtain the database slow query. The payload required by this metric has three parameters: instanceid, region and legend. I want to select instance and region from the drop-down box. The UI corresponding to these two parameters should be a select. I want to enter legend directly. The UI of the legend parameter should be an input box. The UI definition of these parameters is determined by the response message of the metrics API.

If I have different instanceids in different regions, the instanceid list cannot be directly returned by metrics. Therefore, I define an API called "/ options". This interface can dynamically obtain the instanceid list according to the currently selected payload.